### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by Jollywatt**
This is just a few minor edits to docstrings and error messages that had typos or that I felt were inconsistent or opaque, all lumped together into a messy (and maybe opinionated) PR. Feel free to edit!

### CODE REVIEW

File: crates/typst-syntax/src/file.rs [hunk 1]
```diff
@@ -213,25 +213,25 @@ impl FromStr for PackageSpec {
...
+            Err(eco_format!("`{name}` is not a valid package name; expected \"@{{namespace}}/{{name}}:{{version}}\""))?;
         }
 
         s.eat_if(':');
 
         let version = s.after();
         if version.is_empty() {
-            Err("package specification is missing version")?;
+            Err("package specification \"@{namespace}/{name}:{version}\" is missing version")?;
         }
```
Comment (tingerrr) [inline crates/typst-syntax/src/file.rs:235]: You have to adjust the test cases for those, make sure to run `cargo test --workspace` to see how to A) update them if they change, B) if all your changes retained the original behavior.

File: crates/typst-syntax/src/file.rs [hunk 2]
```diff
@@ -285,7 +285,7 @@ impl FromStr for PackageVersion {
             let part = parts
                 .next()
                 .filter(|s| !s.is_empty())
-                .ok_or_else(|| eco_format!("version number is missing {kind} version"))?;
```
Comment (tingerrr) [inline crates/typst-syntax/src/file.rs:288]: Same as above.

File: crates/typst-library/src/compute/foundations.rs [hunk 3]
```diff
@@ -56,14 +56,17 @@ pub fn repr(
...
+/// document) as strings, converting with `repr` if necessary.
+///
 /// # Example
 /// The code below produces the error `panicked with: "this is wrong"`.
 /// ```typ
 /// #panic("this is wrong")
 /// ```
 #[func(keywords = ["error"])]
 pub fn panic(
-    /// The values to panic with.
```
Comment (tingerrr) [inline crates/typst-library/src/compute/foundations.rs:66]: Maybe pedantic, but I think we should keep the `The`.

**Comment by tingerrr** [comment]
Make sure to format the files you edited, too.

**Comment by tingerrr** [comment]
You can mark my review comments as resolved, lgtm!

**Comment by laurmaedje** [comment]
The docs changes are good, but I don't really like the new package error messages. The usage of `{..}` in the error message has no precedent.

**Comment by Jollywatt** [comment]
> The usage of `{..}` in the error message has no precedent.

Fair enough. How about just
```
package specification "@namespace/name:version" is missing namespace
version number "major.minor.patch" is missing patch version
```
instead? I think the messages should definitely hint at the required formats somehow.

**Comment by laurmaedje** [comment]
How about: `version number is missing patch version` and a hint `package specification should be of the form "@namespace/name:version"`. That is, the hint using the hint mechanism instead of being part of the error message. This might need a little restructuring / return type changing from `StrResult` to `HintedStrResult`. 

I'd also prefer to have the error message changes in a separate PR. They are fairly unrelated to the docs changes, which could already be merged.

File: crates/typst/src/eval/func.rs [hunk 4]
```diff
@@ -72,9 +72,14 @@ pub use typst_macros::func;
...
 /// contain positional parameters, named parameters with default values and
-/// [argument sinks]($arguments). The right-hand side of the binding can be a
-/// block or any other expression. It defines the function's return value and
-/// can depend on the parameters.
+/// [argument sinks]($arguments).
+///
+/// The right-hand side of a function binding can be a block or any other
+/// expression. It defines the function's return value and can depend on the
+/// parameters. The `return` keyword can be used inside a block to exit a
+/// function early an return a single value. If no `return` keyword is present,
```
Comment (laurmaedje) [inline crates/typst/src/eval/func.rs:80]:
```suggestion
/// function early and return a single value. If no `return` keyword is present,
```

Comment (laurmaedje) [reply crates/typst/src/eval/func.rs:80]: Also, this is not entirely accurate. Joining also takes place if there is a `return`, but without an explicit value:
```typ
#let f() = {
  [Hello]
  return
  [World]
}

#f() // -> Hello
```
and even
```typ
#let f() = [
  Hello
  #return
  World 
]

#f() // -> Hello
```

File: crates/typst/src/eval/func.rs [hunk 5]
```diff
@@ -72,9 +72,14 @@ pub use typst_macros::func;
...
+/// then like a normal [code block]($scripting/#blocks), the value is the result
```
Comment (laurmaedje) [inline crates/typst/src/eval/func.rs:81]: It isn't really "like a normal code block", it _is_ a normal code block.

File: crates/typst/src/eval/func.rs (hunk 4 above)
Comment (Jollywatt) [reply crates/typst/src/eval/func.rs:80]: Oh, I see. So rather I should say:
> Within a function body, the `return` keyword can be used to exit early. If no explicit return value is given, the body evaluates to the result of joining all expressions preceding the `return`. However, if an explicit value is given, only it is returned.

Comment (laurmaedje) [reply crates/typst/src/eval/func.rs]: I don't think the "However" is necessary, but other than that this is correct and reads well.

**Pull request merged**